lib/auth0/api/authentication_endpoints.rb

```diff
@@ -7,35 +7,48 @@ module AuthenticationEndpoints
       # @param access_token [string] Social provider's access_token
       # @param connection [string] Currently, this endpoint only works for Facebook, Google, Twitter and Weibo
+      # @param scope [string] Use openid to get an id_token, or openid profile email to include user information
+      #   in the id_token. If null, only an access_token will be returned.
+      # @return [json] Returns the access token
+      def obtain_access_token(access_token, connection = 'facebook', scope = 'openid')
+        raise Auth0::InvalidParameter, 'Must supply a valid code' if access_token.to_s.empty?
+        request_params = { client_id: @client_id, access_token: access_token, connection: connection, scope: scope }
+        post('/oauth/access_token', request_params)['access_token']
+      end
+
+      # This is the OAuth 2.0 grant that server processes utilize in order to access an API.
+      # Use this endpoint to directly request an access_token by using the Client Credentials
+      # @see https://auth0.com/docs/api/authentication#client-credentials
+      # @param audience [string] The unique identifier of the target API you want to access.
       # @return [json] Returns the access token
-      def obtain_access_token(access_token = nil, connection = 'facebook', scope = 'openid')
-        if access_token
-          request_params = { client_id: @client_id, access_token: access_token, connection: connection, scope: scope }
-          post('/oauth/access_token', request_params)['access_token']
-        else
-          request_params = { client_id: @client_id, client_secret: @client_secret, grant_type: 'client_credentials' }
-          post('/oauth/token', request_params)['access_token']
-        end
+      def client_credentials(audience = nil)
+        request_params = {
+          client_id:     @client_id,
+          client_secret: @client_secret,
+          grant_type:    'client_credentials',
+          audience:      audience
+        }
+        post('/oauth/token', request_params)['access_token']
       end
 
       # Gets the user tokens using the code obtained through passive authentication in the specified connection
       # @see https://auth0.com/docs/auth-api#!#post--oauth-access_token
       # @param connection [string] Currently, this endpoint only works for Facebook, Google, Twitter and Weibo
       # @param scope [string] Defaults to openid. Can be 'openid name email', 'openid offline_access'
       # @param redirect_uri [string] Url to redirect after authorization
-      # @param redirect_uri [string] The access code obtained through passive authentication
+      # @param code [string] The access code obtained through passive authentication
       # @return [json] Returns the access_token and id_token
-      def obtain_user_tokens(code, redirect_uri, connection = 'facebook', scope = 'openid')
+      def obtain_user_tokens(code, redirect_uri, scope = 'openid')
```

Comment:
I believe the scope is not necessary here since it should have been sent in the first request